src/lib/mdns/Discovery_ImplPlatform.cpp

```diff
@@ -170,9 +178,17 @@ CHIP_ERROR DiscoveryImplPlatform::Advertise(const CommissionAdvertisingParameter
         textEntries[textEntrySize++] = { "VP", reinterpret_cast<const uint8_t *>(vendorProductBuf),
                                          strnlen(vendorProductBuf, sizeof(vendorProductBuf)) };
     }
-#if CHIP_ENABLE_ADDITIONAL_DATA_ADVERTISING
-    textEntries[textEntrySize++] = { "RI", reinterpret_cast<const uint8_t *>(CHIP_ROTATING_DEVICE_ID),
-                                     strlen(CHIP_ROTATING_DEVICE_ID) };
+#if CHIP_ENABLE_ROTATING_DEVICE_ID
+    SuccessOrExit(error =
+                      chip::DeviceLayer::ConfigurationMgr().GetSerialNumber(serialNumber, sizeof(serialNumber), serialNumberSize));
+    SuccessOrExit(error = chip::DeviceLayer::ConfigurationMgr().GetLifetimeCounter(lifetimeCounter));
+    SuccessOrExit(error = AdditionDataPayloadGenerator().generateRotatingDeviceId(
+                      lifetimeCounter, serialNumber, serialNumberSize, rotatingDeviceIdHexBuffer,
+                      ROTATING_DEVICE_ID_MAX_LENGTH * 2 + 1, rotatingDeviceIdHexBufferSize));
+
+    // Rotating Device ID
+    textEntries[textEntrySize++] = { "RI", reinterpret_cast<const uint8_t *>(rotatingDeviceIdHexBuffer),
```

Comment:
could we add some check throughout of textEntrySize that it fits? Amazongly we pre-allocated 4 so it seems enough, however that may not be immediately obvious.

We may want to split this method in general ... it seems to increase a bit too much and have as little bounds check on textEntrySize and subTypeSize.

Reply:
added a size check, let's take these refactoring suggestions in a separate PR